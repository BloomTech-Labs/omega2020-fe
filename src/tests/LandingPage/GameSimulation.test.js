import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom/extend-expect';
import { Typography } from '@material-ui/core';
import Simulation from '../../components/LandingPage/GameSimulation';

configure({ adapter: new Adapter() });

describe('the information on the simulation', () => {
  const simulationRender = renderer.create(<Simulation />); //r represent renderer
  const simulationRoot = simulationRender.root;
  
  const wrapper = mount(<Simulation />);
  const iWrapper = wrapper.instance();
  
  it('snapshot the info on the Game Simulation', () => {
    const simulation = renderer.create(<Simulation />).toJSON();
    expect(simulation).toMatchSnapshot();
  });

  it('checks the root for info', () => {
    expect(wrapper.containsMatchingElement(
    <Typography>
        Learn The Game Play
    </Typography>
    )).toBeTruthy();
    
    expect(wrapper.containsMatchingElement(
    <Typography>
        Start off with something easy
        <br />
        then kick it up a notch
    </Typography>
    )).toBeTruthy();


    expect(iWrapper).toEqual(null)
    expect(simulationRoot.findByProps({id: 'learn'}).props.children).toBeTruthy();
    expect(simulationRoot.findByProps({id: 'start'}).props.children).toBeTruthy();    

  });

});
