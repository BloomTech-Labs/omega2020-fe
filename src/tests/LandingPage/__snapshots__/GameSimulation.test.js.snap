// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the information on the simulation snapshot the info on the Game Simulation 1`] = `
<div
  className="MuiBox-root MuiBox-root-45 makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-46 makeStyles-mainContainer-2"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <div
          className="MuiBox-root MuiBox-root-47"
        >
          <div
            className="MuiBox-root MuiBox-root-48"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
              id="learn"
              style={
                Object {
                  "fontSize": "1.3rem",
                }
              }
            >
              Learn The Game Play
            </p>
            <br />
            <p
              className="MuiTypography-root makeStyles-subheader-3 MuiTypography-body1"
              id="start"
            >
              Start off with something easy
              <br />
              then kick it up a notch
            </p>
            <br />
            <br />
            <br />
            <br />
          </div>
        </div>
      </div>
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <div
          className="MuiBox-root MuiBox-root-49 makeStyles-Simulation-4"
        >
          <div
            className="MuiBox-root MuiBox-root-50 makeStyles-root-10"
          >
            <div
              className="MuiBox-root MuiBox-root-51 makeStyles-mainContainer-11"
            >
              <div
                className="MuiBox-root MuiBox-root-52"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <p
                  className="MuiTypography-root MuiTypography-body1"
                  id="solve"
                  style={
                    Object {
                      "fontSize": "1.3rem",
                    }
                  }
                >
                  Try to solve this 4x4 sudoku:
                </p>
                <br />
                <p
                  className="MuiTypography-root makeStyles-subheader-12 MuiTypography-body1"
                  id="row"
                >
                  Each row, column and square needs to be filled
                  <br />
                  out with the numbers 1-4, without repeating any
                  <br />
                  numbers within the row, column or square.
                </p>
                <br />
                <div
                  className="MuiBox-root MuiBox-root-53"
                >
                  <img
                    alt="Game4x4"
                    src="Game4x4.png"
                    width="300"
                  />
                </div>
              </div>
            </div>
          </div>
          <br />
          <div
            className="MuiBox-root MuiBox-root-54 makeStyles-root-21"
          >
            <div
              className="MuiBox-root MuiBox-root-55 makeStyles-mainContainer-22"
            >
              <div
                className="MuiBox-root MuiBox-root-56"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <p
                  className="MuiTypography-root MuiTypography-body1"
                  id="solve"
                  style={
                    Object {
                      "fontSize": "1.3rem",
                    }
                  }
                >
                  Try to solve this 6x6 sudoku:
                </p>
                <br />
                <p
                  className="MuiTypography-root makeStyles-subheader-23 MuiTypography-body1"
                  id="row"
                >
                  Each row, column and square needs to be filled
                  <br />
                  out with the numbers 1-6, without repeating any
                  <br />
                  numbers within the row, column or square.
                </p>
                <br />
                <div
                  className="MuiBox-root MuiBox-root-57"
                >
                  <img
                    alt="Game6x6"
                    src="Game6x6.png"
                    width="300"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiCardHeader-content"
      />
    </div>
  </div>
</div>
`;
